subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/UnsafeConfigurationResolutionDeprecationIntegrationTest.groovy

```diff
@@ -135,12 +104,11 @@ class UnsafeConfigurationResolutionDeprecationIntegrationTest extends AbstractDe
         """
 
         when:
-        executer.expectDeprecationWarning()
         executer.withArgument("--parallel")
-        succeeds(":bar:help")
+        fails(":bar:help")
 
         then:
-        outputContains("The configuration :baz:baz was resolved without accessing the project in a safe manner.")
+        failureCauseContains("The configuration :baz:baz was resolved from a context different than the project context.")
```

Comment:
This is better but I think we can do even better. Something like this:

`The configuration `:baz:baz` has been resolved, but this was done outside of the project itself. In order for cross-project publications to be safe, configurations should only be resolved within the project which owns them. Please refer to the documentation at ....`.

Reply:
I now realize that neither "was resolved" nor "has been resolved" work, since we fail the build now when attempted.

And thanks for thinking about documentation. We indeed have https://docs.gradle.org/6.8.1/userguide/viewing_debugging_dependencies.html#sub:resolving-unsafe-configuration-resolution-errors which needs to be updated.

Will rework the message and propose something.

Reply:
New proposal:

> Resolution of the configuration :bar:bar was attempted from a context different than the project context. Have a look at the documentation to understand why this is a problem and how it can be resolved.
  https://docs.gradle.org/7.0-20210203230000+0000/userguide/viewing_debugging_dependencies.html#sub:resolving-unsafe-configuration-resolution-errors